presto-main/src/main/java/com/facebook/presto/execution/buffer/BroadcastOutputBuffer.java

```diff
@@ -400,4 +436,16 @@ OutputBufferMemoryManager getMemoryManager()
     {
         return memoryManager;
     }
+
+    private void dereferencePage(SerializedPage pageSplit, Lifespan lifespan)
+    {
+        long outstandingPageCount = outstandingPageCountPerLifespan.get(lifespan).decrementAndGet();
+        if (outstandingPageCount == 0 && noMorePagesForLifespan.contains(lifespan)) {
+            checkState(lifespanCompletionCallback != null);
+            lifespanCompletionCallback.accept(lifespan);
+        }
+
+        checkState(totalBufferedPages.decrementAndGet() >= 0);
```

Comment:
Add a message in case it ever fails. (include the incorrect `totalBufferedPages` in the message)

Reply:
These are refactored code and not related to this lifespan completion change . We can do them in a separate refactor PR :)